test/unit/test_util.c

```diff
@@ -113,6 +113,42 @@ bool test_autonames(void) {
 	mu_end;
 }
 
+bool test_file_slurp(void) {
+
+#ifdef __WINDOWS__
+#define S_IRWXU _S_IREAD | _S_IWRITE
+#endif
+
+	char* test_file =  "./empty_file";
+	size_t s;
+	char* content;
+	char* some_words = "some words";
+
+	close(open(test_file, O_CREAT, S_IRWXU));
+
+	content = r_file_slurp(test_file, &s);
+
+	mu_assert_eq (s, 0, "size should be zero");
+	mu_assert_eq (strlen(content), 0, "returned buffer should be empty");
```

Comment:
well. assuming the file is zero, the buffer shuold not contain anything, not even a null byte, because files are binary data not text. But i can get this assumption could be "handy" for some usages of the api

Reply:
There are many cases where `r_file_slurp` is used assuming that it returns a null-terminated string. This is why I suggested this behaviour (which by the way already always holds except when `sz == 0`)